lib/ansible/module_utils/basic.py

```diff
@@ -1879,10 +1879,17 @@ def _load_params(self):
 
     def _log_to_syslog(self, msg):
         if HAS_SYSLOG:
-            module = 'ansible-%s' % self._name
-            facility = getattr(syslog, self._syslog_facility, syslog.LOG_USER)
-            syslog.openlog(str(module), 0, facility)
-            syslog.syslog(syslog.LOG_INFO, msg)
+            try:
+                module = 'ansible-%s' % self._name
+                facility = getattr(syslog, self._syslog_facility, syslog.LOG_USER)
+                syslog.openlog(str(module), 0, facility)
+                syslog.syslog(syslog.LOG_INFO, msg)
+            except TypeError as e:
+                self.fail_json(
```

Comment:
I feel, rather failing we should warn user about this and continue, which will atleast continue the run.

Reply:
I thought about it for quite a while and I think if the user has logging enabled, it's important that we log, because it might be used for auditing and so on. If we promise to log something and can't, we probably shouldn't do that thing and should fail out, IMHO.

Reply:
logging is turned on by default for modules, so i would not make it fatal, also adding a warning allows user to 'log' this at the controller.

Reply:
It doesn't seem we are hitting this very often, but for the times we do, I also don't think it should be fatal. Since our solution is "turn off logging" I'm not sure failing is more useful than warning that we couldn't log something and continuing.